src/components/Video/Video.js

```diff
@@ -0,0 +1,153 @@
+import Hls from 'hls.js';
+import React, { useEffect, useRef, useState } from 'react';
+import { useBeforeUnload } from 'react-router-dom';
+
+import { useKeyDown } from '../../../hooks/useKeyDown';
+import { KEY_PLAYBACK_RATE } from '../../../utils/constants';
+
+import VolumeLogo from './img/volume.svg';
+
+import styles from './Video.scss';
+
+const Video = ({ id, isCardVideo = false, videoLink, previewImageLink }) => {
+  const [showPreview, setShowPreview] = useState(true);
+  const [isMutedVideo, setIsMutedVideo] = useState(isCardVideo);
+  const [canPlay, setCanPlay] = useState(true);
+  const video = useRef(null);
+
+  const changePlaybackRate = (pressedKey) => {
+    switch (pressedKey) {
+      case 'u':
+        if (video.current.playbackRate > 0.25) {
+          video.current.playbackRate -= 0.25;
+        }
+        break;
+
+      case 'i':
+        if (video.current.playbackRate < 4) {
+          video.current.playbackRate += 0.25;
+        }
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  useKeyDown((pressedKey) => {
+    changePlaybackRate(pressedKey);
+  }, KEY_PLAYBACK_RATE);
+
+  const saveVideoProgress = (video, id) => {
+    if (id) localStorage.setItem(id, video.currentTime);
+  };
+
+  useBeforeUnload(() => {
+    saveVideoProgress(video.current, id);
+  });
+
+  const stopVideo = () => {
+    if (canPlay) setShowPreview((prevState) => !prevState);
```

Comment:
Although there is only one line, but for better readability it is better if (canPlay) {
setShowPreview((prevState) => !prevState)
};